src/lib/mongoBackend/mongoUnsubscribeContextAvailability.cpp

```diff
@@ -76,45 +76,57 @@ HttpStatusCode mongoUnsubscribeContextAvailability
   if (!safeGetSubId(requestP->subscriptionId, &id, &(responseP->statusCode)))
   {
     reqSemGive(__FUNCTION__, "ngsi9 unsubscribe request (safeGetSubId fail)", reqSemTaken);
+
     if (responseP->statusCode.code == SccContextElementNotFound)
     {
       // FIXME: doubt: invalid OID format? Or, subscription not found?
       std::string details = std::string("invalid OID format: '") + requestP->subscriptionId.get() + "'";
       alarmMgr.badInput(clientIp, details);
     }
-    else // SccReceiverInternalError
+    else  // SccReceiverInternalError
     {
       LM_E(("Runtime Error (exception getting OID: %s)", responseP->statusCode.details.c_str()));
     }
+
     return SccOk;
   }
 
   if (!collectionFindOne(getSubscribeContextAvailabilityCollectionName(tenant), BSON("_id" << id), &sub, &err))
   {
     reqSemGive(__FUNCTION__, "ngsi9 unsubscribe request (mongo db exception)", reqSemTaken);
     responseP->statusCode.fill(SccReceiverInternalError, err);
+
     return SccOk;
   }
+
   alarmMgr.dbErrorReset();
 
   if (sub.isEmpty())
   {
     responseP->statusCode.fill(SccContextElementNotFound);
     reqSemGive(__FUNCTION__, "ngsi9 unsubscribe request (no subscriptions)", reqSemTaken);
+
     return SccOk;
   }
 
   /* Remove document in MongoDB */
+
+  //
   // FIXME: I would prefer to do the find and remove in a single operation. Is the some similar
-  // to findAndModify for this?  
-  if (!collectionRemove(getSubscribeContextAvailabilityCollectionName(tenant), BSON("_id" << OID(requestP->subscriptionId.get())), &err))
+  // to findAndModify for this?
+  //
+
+  std::string colName = getSubscribeContextAvailabilityCollectionName(tenant);
```

Comment:
As discussed in our last talk, being a "should" rule we don't need to apply to everyplace, although probably it would be a good idea in all the cases we use BSON() macro in a non trivial way (e.g. trivial way `BSON ("_id" << id)`).